File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if prices == []: return 0
        mini = prices[0]
        maxP = float('-inf')
        for ele in prices:
            maxP = max(maxP, ele - mini)
            mini = min(ele, mini)
            
        return maxP
            
        

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int profit = 0;
        
        int minVal = INT_MAX;
        
        for(auto price : prices){
            minVal = min(price, minVal);
            profit = max(profit, price - minVal);
        }
        
        
        return profit;
        
    }
};